src/scripts/extensions/userDataBoundaryHelper.ts

```diff
@@ -39,7 +40,21 @@ export class UserDataBoundaryHelper {
 			let domainValue = userInfo.emailAddress.substring(
 				userInfo.emailAddress.indexOf("@") + 1);
 			const urlDataBoundaryDomain: string = UrlUtils.addUrlQueryValue(Constants.Urls.userDataBoundaryDomain, Constants.Urls.QueryParams.domain, domainValue);
-			HttpWithRetries.get(urlDataBoundaryDomain).then((response: Response) => {
+			const headers = {
+				/**
+				 * TODO: Update the officeApplicationNumber to the correct value
+				 * once OneNote Web Clipper is onboarded to HRD.
+				 */
+				"X-Office-Application": undefined,
+				"Enlightened-Hrd-Client": "1", // To turn on the EUDB/Telemetry region
+				"X-Office-Platform": "Web",
+				"Content-Type": "application/json",
+			};
+			const correlationId = StringUtils.generateGuid();
```

Comment:
Just for my information, what is this correlation id used for, since we are generating a new one everytime?

Reply:
As per https://www.owiki.ms/wiki/ICE_Services/Email_HRD, the correlation id is a unique GUID which will be used by the Email HRD team in their logs. If we ever need them to look-up their logs, we can give them a correlation id and a timestamp (preferably in UTC), and they can look up the logs.

It would have been best if we could have used the session id here. However, on using `Clipper.getUserSessionIdWhenDefined` here, clipper goes into an infinite loop. Hence, I have followed what ToDo does [here](https://dev.azure.com/outlookweb/Outlook%20Web/_git/to-do-web?path=/packages/todo-core/service/HRDService.ts&version=GBmaster&line=30&lineEnd=30&lineStartColumn=3&lineEndColumn=39&lineStyle=plain&_a=contents). We can re-visit this logic if there is ever a need for us to actually check the Email HRD logs.

Reply:
In the code u have linked above, I see that the correlation id is getting logged hence in case of any failure or such we would be able to query it, but in our case that won't be possible since we are not logging it anywhere. If it's possible to log it, let's do that or else we can add a TODO and revisit it later as you mentioned.